src/dev.c

```diff
@@ -195,26 +273,87 @@ fido_init(int flags)
 {
 	if (flags & FIDO_DEBUG || getenv("FIDO_DEBUG") != NULL)
 		fido_log_init();
+#if defined(LINUX)
+#ifdef USE_HIDAPI
+	fido_dev_register_manifest_func(hidapi_dev_info_manifest);
+	if (hid_init() != 0)
+		errx(1, "hid_init failed");
+#else
+	fido_dev_register_manifest_func(fido_dev_info_manifest_linux);
+#endif // USE_HIDAPI
+#elif defined(OSX)
+	fido_dev_register_manifest_func(fido_dev_info_manifest_osx);
+#elif defined(WIN)
+	fido_dev_register_manifest_func(fido_dev_info_manifest_win);
+#elif defined(OPENBSD)
+	fido_dev_register_manifest_func(fido_dev_info_manifest_openbsd);
+#endif // LINUX|OSX|WIN|OPENBSD
+}
+
+void
+fido_exit()
+{
+#ifdef USE_HIDAPI
+	hid_exit();
+#endif
+	dev_manifest_func_node_t *curr = manifest_funcs;
+	dev_manifest_func_node_t *next = NULL;
+	while (curr != NULL) {
+		next = curr->next;
+		free(curr);
+		curr = next;
+	}
+	manifest_funcs = NULL;
```

Comment:
This should probably happen elsewhere. We may need to change the abstraction a bit, perhaps by introducing a function that does the reverse of fido_dev_register_manifest_func().